src/catalog.c

```diff
@@ -444,6 +449,42 @@ catalog_table_next_seq_id(Catalog *catalog, CatalogTable table)
 	return DatumGetInt64(DirectFunctionCall1(nextval_oid, ObjectIdGetDatum(relid)));
 }
 
+/*
+ * Become the user that owns the catalog schema.
+ *
+ * This might be necessary for users that do operations that require changes to
+ * the catalog.
+ *
+ * The caller should pass a CatalogSecurityContext where the current security
+ * context will be saved. The original security context can later be restored
+ * with catalog_restore_user().
+ */
+bool
+catalog_become_owner(CatalogPrefix *catalog_prefix, CatalogSecurityContext *sec_ctx)
+{
+	GetUserIdAndSecContext(&sec_ctx->saved_uid, &sec_ctx->saved_security_context);
+
+	if (sec_ctx->saved_uid != catalog_prefix->owner_uid)
+	{
+		SetUserIdAndSecContext(catalog_prefix->owner_uid,
```

Comment:
I didn't change this function at all btw, just moved it around so that it's more clear what kind of function it is. What do you mean by "change back here"?